### DISCUSSION TITLE
Access $page before create app

### DISCUSSION THREAD

**Discussion by lucraraujo** [inertiajs/inertia#996] [Help (Vue)]
I'm trying to use an ACL plugin in my application (https://github.com/mblarsen/vue-browser-acl) and it requires to get the user before the application is created. Any ideas?

```js
//app.js
import Vue from 'vue'
import { createInertiaApp, Link } from '@inertiajs/inertia-vue'
import Acl from 'vue-browser-acl'
import { rules, user } from '@/Acl'

Vue.use(Acl, user, rules)

createInertiaApp({
  resolve: name => import(`./Pages/${name}.vue`).then(({ default: page }) => {
    page.layout = Layout
    return page
  }),
  setup ({ el, app, props }) {
    new Vue({
      render: h => h(app, props)
    }).$mount(el)
  }
})
```

```js
//Acl.js
const user = this.$page.props.auth.user

const rules = (acl) => {
  acl.rule('admin', (user) => user.roles.includes('ADMIN'))
}

export {
  user,
  rules
}
```

**Comment by ajnsn** [comment]
Hey @lucraraujo 
I do not know the package but here are some possible tips. 

Looking at the [setup section](https://github.com/mblarsen/vue-browser-acl#setup) it lools like that an asynchronous retrieved user object is supported. 

```js
const user = () => store.auth.user;
```

Note the function. This is most likely a Vuex example but you might easily be able to adopt it for Inertia.

Another option could be to injecting the initial application user as `data-user`, together with the `data-page`. Or like the above setup link shows pass the user with something like `window.__INITIAL_STATE__` in a `<script>` tag within your `app.blade.php`

But what I would rather recommend is to build your own approach, it's very easy. This could be a start:
https://mmccaff.github.io/2018/11/03/laravel-permissions-in-vue-components/
